src/Findologic/Api/FindologicSearchService.php

```diff
@@ -213,4 +212,16 @@ protected function isCategoryPage(NavigationRequestHandler $handler, ProductList
 
         return !empty($isCategoryPage);
     }
+
+    protected function disableFindologicService(ProductListingCriteriaEvent $event): void
+    {
+        /** @var FindologicService|null $findologicService */
+        $findologicService = $event->getContext()->getExtension('findologicService');
+        if (!$findologicService) {
+            $findologicService = new FindologicService();
+            $event->getContext()->addExtension('findologicService', $findologicService);
+        }
+
+        $findologicService->disable();
```

Comment:
Why do we need to call disable when the findologicService doesn't exist? Wouldn't this already be "disabled"?

Reply:
In the usual case it will be gotten from the context. Creating a new instance is really a corner case, as a plugin explicitly would need to call this. At this point I do not really care where the `FindologicService` comes from, and in which state it currently is, I just want to disable it.

Reply:
> as a plugin explicitly would need to call this

Why does another plugin need to call this, seems wrong to me?!

Further we have several usages of `$event->getContext()->getExtension('findologicService')`, will those work fine without this check?

![Screen Shot 2021-02-10 at 18 08 51](https://user-images.githubusercontent.com/511559/107545050-20ef0300-6bcb-11eb-923e-e82ab278ef8b.png)

Reply:
Issue here is the whole stack doesn't get executed (e.g. `FrontendSubscriber`). In order for your described code to actually work, there already must be an existing `FindologicService`, which must be active. Therefore disabling it when the request fails or the alivetest fails, is the proper way to deal with this.